### PR TITLE
[#1669] fixed pins position style

### PR DESCRIPTION
**Pull request by aluarius**
close: #1669 

## Description
Fixed pins position style. The initial problem is caused by some props combinations that  weren't processed.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-slider/VaSlider.vue [hunk 1]
```diff
@@ -515,9 +515,21 @@ export default defineComponent({
...
+    const pinPositionStep = computed(() => {
+      if (props.min === 0 && props.max === 100 && props.step !== 1) {
+        // only step prop passed
+        return props.step
+      } else if ((props.min !== 0 || props.max !== 100) && props.step === 1) {
+        // only min or/and max prop passed
+        return (100 / (props.max - props.min))
+      } else {
+        // step and (min or max) prop passed
+        return ((props.step) * 100) / (props.max - props.min)
```
Comment (m0ksem) [inline packages/ui/src/components/va-slider/VaSlider.vue:527]: Can't we just use this last formula?

Comment (aluarius) [reply packages/ui/src/components/va-slider/VaSlider.vue:524]: Yes, you're right. There was another formula before, but now (taking into account, that step can be === 1 in the second case), they're the same.

File: packages/ui/src/components/va-slider/VaSlider.vue [hunk 2]
```diff
@@ -515,9 +515,10 @@ export default defineComponent({
       }
     }
 
+    const pinPositionStep = computed(() => ((props.step) * 100) / (props.max - props.min))
     const getPinStyles = (pin: number) => ({
       backgroundColor: checkActivePin(pin) ? getColor(props.color) : getHoverColor(getColor(props.color)),
-      [pinPositionStyle.value]: `${pin * props.step}%`,
+      [pinPositionStyle.value]: `${pin * pinPositionStep.value}%`,
       transition: hasMouseDown.value ? 'none' : 'background-color .3s ease-out .1s',
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-slider/VaSlider.vue:522]: `'background-color .3s ease-out .1s'` better move it to CSS variable.

**Review by rustem-nasyrov — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/va-slider/VaSlider.vue [hunk 3]
```diff
@@ -515,9 +515,10 @@ export default defineComponent({
       }
     }
 
+    const pinPositionStep = computed(() => ((props.step) * 100) / (props.max - props.min))
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-slider/VaSlider.vue:518]:
```suggestion
    const pinPositionStep = computed(() => (props.step * 100) / (props.max - props.min))
```

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]
Pay attention to code line #326. I guess there is a mistake.

File: packages/ui/src/components/va-slider/VaSlider.vue (hunk 3 above)
Comment (m0ksem) [reply packages/ui/src/components/va-slider/VaSlider.vue:518]: IMHO, the percent formula (from school) looks like `part / whole * 100`, so my suggestion here ✌:

```suggestion
    const pinPositionStep = computed(() => (props.step / (props.max - props.min) * 100) 
```

File: packages/ui/src/components/va-slider/_variables.scss [hunk 4]
```diff
@@ -12,6 +12,9 @@
   --va-slider-track-transition: 0.5s ease-out;
   --va-slider-track-opacity: 1;
 
+  /* Pin */
+  --va-slider-pin-transition: 0.3s ease-out 0.1s;
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-slider/_variables.scss:16]:
```suggestion
  --va-slider-pin-transition: background-color 0.3s ease-out 0.1s;
```

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]

**Comment by aluarius** [comment]
> Pay attention to code line 326. I guess there is a mistake.

Yes, it may be related to https://github.com/epicmaxco/vuestic-ui/issues/1673, but it will be solved separately from this pr.

**Review by LighthouseKeeperYN — APPROVED** [review]

**Review by rustem-nasyrov — APPROVED** [review]

**Review by RVitaly1978 — APPROVED** [review]

**Pull request merged**